btree/btree.c

```diff
@@ -3642,21 +3654,24 @@ static int64_t btree_get_tick(struct m0_sm_op *smop)
 			return P_SETUP;
 	case P_SETUP:
 		bop->bo_arbor->t_height = tree->t_height;
-		oi = level_alloc(tree->t_height);
-		if (oi == NULL)
+		level_alloc(oi, bop->bo_arbor->t_height);
+		if (oi->i_level == NULL)
 			return fail(bop, M0_ERR(-ENOMEM));
 		return P_LOCKALL;
 	case P_LOCKALL:
 		if (bop->bo_flags & BOF_LOCKALL)
-			/* return m0_sm_op_sub(&bop->bo_op, P_LOCK, P_DOWN); */
-			return P_LOCK;
+			return lock_op_init(&bop->bo_op, &bop->bo_i->i_nop,
+				            bop->bo_arbor->t_desc, P_DOWN);
 		/** Fall through if LOCKALL flag is not set. */
-	case P_DOWN:
+	case P_DOWN: {
+		bool lock_acquired = bop->bo_flags & BOF_LOCKALL;
```

Comment:
You don't have to assign lock_acquired at each level. Just define at the top of the function. Initially it will be zero. If it is updated in P_LOCKALL , it will be returned from state machine and then you will have updated lock_acquired next time onwards. Likewise in other state machines.

Reply:
we can have lock_acquired at top of function but we can not initialized with zero . because it is local for function ,even if we  update flag LOCKALL , it will be again zero for NEXTDOWN state. rather we can initialized flag with bop->bo_flags & BOF_LOCKALL

Reply:
made respective changes